go/libkb/bootstrap_active_device.go

```diff
@@ -78,23 +88,28 @@ func bootstrapActiveDeviceReturnRawError(m MetaContext, uid keybase1.UID, device
 		m.Debug("active device is current")
 		return nil
 	}
-	uv, sib, sub, deviceName, err := LoadUnlockedDeviceKeys(m, uid, deviceID, online, false)
+	mode := LoadUnlockedDeviceKeysModeNormal
+	if !online {
+		mode = LoadUnlockedDeviceKeysModeOffline
+	}
+	uv, sib, sub, deviceName, err := LoadUnlockedDeviceKeys(m, uid, deviceID, mode)
 	if err != nil {
 		return err
 	}
 	err = m.SetActiveDevice(uv, deviceID, sib, sub, deviceName, KeychainModeOS)
 	return err
 }
 
-func LoadUnlockedDeviceKeys(m MetaContext, uid keybase1.UID, deviceID keybase1.DeviceID, online bool, staleOK bool) (uv keybase1.UserVersion, sib GenericKey, sub GenericKey, deviceName string, err error) {
+func LoadUnlockedDeviceKeys(m MetaContext, uid keybase1.UID, deviceID keybase1.DeviceID, mode LoadUnlockedDeviceKeysMode) (uv keybase1.UserVersion, sib GenericKey, sub GenericKey, deviceName string, err error) {
 	defer m.Trace("LoadUnlockedDeviceKeys", func() error { return err })()
 
 	// use the UPAKLoader with StaleOK, CachedOnly in order to get cached upak
 	arg := NewLoadUserArgWithMetaContext(m).WithUID(uid).WithPublicKeyOptional()
-	if !online {
+
+	if mode == LoadUnlockedDeviceKeysModeOffline {
```

Comment:
make this a switch

Reply:
I thought the new rule was only use a switch if there's a `default` case.